yescrypt/src/lib.rs

```diff
@@ -164,23 +142,22 @@ pub fn yescrypt_kdf(
 }
 
 unsafe fn yescrypt_kdf_inner(
-    mut shared: *const Shared,
-    mut local: *mut Local,
+    local: *mut Local,
     mut passwd: *const uint8_t,
     mut passwdlen: size_t,
-    mut salt: *const uint8_t,
-    mut saltlen: size_t,
-    mut params: *const Params,
-    mut buf: *mut uint8_t,
-    mut buflen: size_t,
+    salt: *const uint8_t,
+    saltlen: size_t,
+    params: *const Params,
+    buf: *mut uint8_t,
+    buflen: size_t,
 ) -> libc::c_int {
-    let mut flags: Flags = (*params).flags;
-    let mut N: uint64_t = (*params).N;
-    let mut r: uint32_t = (*params).r;
-    let mut p: uint32_t = (*params).p;
-    let mut t: uint32_t = (*params).t;
-    let mut g: uint32_t = (*params).g;
-    let mut NROM: uint64_t = (*params).NROM;
+    let flags: Flags = (*params).flags;
+    let N: uint64_t = (*params).N;
+    let r: uint32_t = (*params).r;
```

Comment:
For future reference all the superfluous syntax here is easy to get rid of:

```suggestion
    let flags = params.flags;
    let N = params.N;
    let r = params.r;
```

Not a blocker for this PR though, you're already doing a lot already!